### ISSUE TITLE
Try catch IIFE causes false positive on unreachable code

### ISSUE THREAD

**Issue by ycmjason** [microsoft/TypeScript#37789]
**TypeScript Version:**  3.8.3

**Search Terms:**
try, catch, unreachable, false positive

**Code**

```ts
const f = () => {
  const x = (() => {
    try {
      return 3
    } catch(e) {
      return null;
    }
  })();

  return 3
};
```

**Expected behavior:**
The last line in `f` should be reachable.

```ts
return 3
```

**Actual behavior:**

Shown as unreachable code

**Playground Link:**
https://www.typescriptlang.org/play?#code/MYewdgzgLgBAZjAvDAFASiQPhgbwFAwyiSwAeSq6WuBhMUATgJ4110MCmUArg2DAGZahAL5EAhlGAALFBwz42hTjz4ww3ADaaA3MJgjaItOj20VvfkJF6gA

I believe this issue do not exist in 3.7. Tested on playground.

**Comment by IllusionMH** [comment]
Already fixed
https://github.com/microsoft/TypeScript/issues/36828
